components/patient-data/impl/src/main/java/org/phenotips/data/internal/PatientDeathdateUpdater.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.internal;
+
+import org.phenotips.Constants;
+import org.phenotips.data.Patient;
+import org.phenotips.data.events.PatientChangingEvent;
+
+import org.xwiki.component.annotation.Component;
+
+import org.xwiki.container.Container;
+import org.xwiki.container.Request;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.Event;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Removes the date of death object iff "unknown date of death" checkbox is checked.
+ *
+ * @version $Id$
+ */
+@Component
+@Named("patient-deathdate-updater")
+@Singleton
+public class PatientDeathdateUpdater extends AbstractEventListener
+{
+    /** Needed for getting access to the request. */
+    @Inject
+    private Container container;
+
+    /** Default constructor, sets up the listener name and the list of events to subscribe to. */
+    public PatientDeathdateUpdater()
+    {
+        super("patient-deathdate-updater", new PatientChangingEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        XWikiDocument doc = (XWikiDocument) source;
+
+        BaseObject patientRecordObj = doc.getXObject(Patient.CLASS_REFERENCE);
+        if (patientRecordObj == null) {
+            return;
+        }
+        boolean unknownDeathDate = getParameter("date_of_death_unknown", patientRecordObj.getNumber());
```

Comment:
Definitely a good idea. This was done this way basically because I did not know how/when exactly onEvent hooks work, and just copied the code from another updater (BirthDateUpdater - should we update that one in a similar manner as well?)